### PR TITLE
Improve unit-tests coverage

### PR DESCRIPTION
**Pull request by Tal-or**
This PR is intended to improve our unit tests coverage.
Unit tests were added to the following packages:
```pkg/controllers```
```pkg/apply```
```pkg/objectstate/sched```
```pkg/status```

### CODE REVIEW

File: pkg/numaresourcesscheduler/objectstate/sched/sched_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		if podSpec.Volumes[0].Name != SchedulerConfigMapVolumeName {
+			t.Errorf("failed to update deployemt volume name, expected: %q actual: %q", SchedulerConfigMapVolumeName, podSpec.Volumes[0].Name)
+		}
+		if podSpec.Volumes[0].ConfigMap.LocalObjectReference.Name != tc.cmName {
+			t.Errorf("failed to update deployemt volume configmap name, expected: %q actual: %q", tc.cmName, podSpec.Volumes[0].ConfigMap.LocalObjectReference.Name)
+		}
+	}
+}
+
+func TestDeploymentNamespacedNameFromObject(t *testing.T) {
```
Comment (yanirq) [inline pkg/numaresourcesscheduler/objectstate/sched/sched_test.go:89]: non compulsory but perhaps the use of more meaningful variables will make the test easy to read (dp, nname)

Comment (Tal-or) [reply pkg/numaresourcesscheduler/objectstate/sched/sched_test.go:89]: ```dp``` is an abbreviate for deployment and ```nname``` for namespaced name. it's quite common actually and I saw that in used in multiple places

**Comment by Tal-or** [comment]
@cynepco3hahue do you have permission to retest the format job?

File: controllers/numaresourcesscheduler_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
+package controllers
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```
Comment (fromanirh) [inline controllers/numaresourcesscheduler_controller_test.go:6]: not a fan of this style of imports, but I guess we can clean up later

File: controllers/numaresourcesscheduler_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	nrsv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	schedmanifests "github.com/openshift-kni/numaresources-operator/pkg/numaresourcesscheduler/manifests/sched"
+	"github.com/openshift-kni/numaresources-operator/pkg/status"
+	"github.com/openshift-kni/numaresources-operator/pkg/testutils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
```
Comment (fromanirh) [inline controllers/numaresourcesscheduler_controller_test.go:20]: the order of imports is a bit messed up, please clean up if you resubmit

File: controllers/numaresourcesscheduler_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
+package controllers
```
Comment (fromanirh) [inline controllers/numaresourcesscheduler_controller_test.go:1]: missing copyright boilerplate

File: doc/examples/sched.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 apiVersion: nodetopology.openshift.io/v1alpha1
 kind: NUMAResourcesScheduler
 metadata:
-  name: topo-aware-scheduler
+  name: numaresourcesscheduler
```
Comment (fromanirh) [inline doc/examples/sched.yaml:4]: not sure we want to enforce this name, on one hand this is likely to clash with the name we will use when we switch to SSOP. OTOH, this seems also a fair thing to change.

File: controllers/numaresourcesscheduler_controller_test.go (hunk 2 above)
Comment (Tal-or) [reply controllers/numaresourcesscheduler_controller_test.go:6]: This is usually how we import them.
No biggy I'll use explicit names.

Comment (fromanirh) [reply controllers/numaresourcesscheduler_controller_test.go:6]: > This is usually how we import them. No biggy I'll use explicit names.

no worries for now, we can keep them like that and change everything in a single sweep with a later PR.

File: doc/examples/sched.yaml (hunk 5 above)
Comment (Tal-or) [reply doc/examples/sched.yaml:4]: IIRC, SSOP using the ```secondary-scheduler``` name for its CR

**Comment by Tal-or** [comment]
@fromanirh Any additional comments here?

File: controllers/numaresourcesscheduler_controller.go [hunk 6]
```diff
@@ -40,6 +41,11 @@ import (
 	"github.com/openshift-kni/numaresources-operator/pkg/status"
 )
 
+const (
+	defaultNUMAResourcesSchedulerCrName                      = "numaresourcesscheduler"
```
Comment (fromanirh) [inline controllers/numaresourcesscheduler_controller.go:45]: this doesn't seem needed but it also doesn't seem wrong, and allow us to test a bit better, so let's keep it

File: doc/examples/sched.yaml (hunk 5 above)
Comment (fromanirh) [reply doc/examples/sched.yaml:4]: let's keep like this for now; the biggest reason is we gave up to be 1:1 compatible with SSOP, so no point in using the same name: it only adds confusion.

File: pkg/apply/apply_test.go [hunk 7]
```diff
@@ -0,0 +1,93 @@
+package apply
```
Comment (fromanirh) [inline pkg/apply/apply_test.go:1]: (possibly) missing copyright boilerplate

File: pkg/apply/apply_test.go [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+import (
+	"context"
+	"github.com/google/go-cmp/cmp"
```
Comment (fromanirh) [inline pkg/apply/apply_test.go:5]: mixed import order (stdlib and 3rd party)

File: pkg/apply/apply_test.go [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+			objectState: objectstate.ObjectState{
+				Existing: dsExist,
+				Desired:  dsDesired,
+				Compare:  compare.Object,
+				Merge:    merge.MetadataForUpdate,
+			},
+		},
+	}
+
+	for _, ts := range testCases {
```
Comment (fromanirh) [inline pkg/apply/apply_test.go:83]: nice!

File: pkg/numaresourcesscheduler/objectstate/sched/sched_test.go [hunk 10]
```diff
@@ -0,0 +1,98 @@
+package sched
```
Comment (fromanirh) [inline pkg/numaresourcesscheduler/objectstate/sched/sched_test.go:1]: (possibly) missing copyright boilerplate

File: pkg/numaresourcesscheduler/objectstate/sched/sched_test.go [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+import (
+	nrsv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
```
Comment (fromanirh) [inline pkg/numaresourcesscheduler/objectstate/sched/sched_test.go:4]: messed import order (stdlib, kube, 3rd party)

File: pkg/numaresourcesscheduler/objectstate/sched/sched_test.go [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+	}
+
+	podSpec := &dp.Spec.Template.Spec
+	for _, tc := range testCases {
+		UpdateDeploymentConfigMapSettings(dp, tc.cmName)
+		if podSpec.Volumes[0].Name != SchedulerConfigMapVolumeName {
+			t.Errorf("failed to update deployemt volume name, expected: %q actual: %q", SchedulerConfigMapVolumeName, podSpec.Volumes[0].Name)
+		}
+		if podSpec.Volumes[0].ConfigMap.LocalObjectReference.Name != tc.cmName {
+			t.Errorf("failed to update deployemt volume configmap name, expected: %q actual: %q", tc.cmName, podSpec.Volumes[0].ConfigMap.LocalObjectReference.Name)
```
Comment (fromanirh) [inline pkg/numaresourcesscheduler/objectstate/sched/sched_test.go:84]: typo: deployment

**Review by fromanirh — COMMENTED** [review]
few minor things and we are done

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**